src/Components/OATPropertyEditor/Internal/TemplateListItem.tsx

```diff
@@ -0,0 +1,93 @@
+import React, { useState } from 'react';
+import { Text, IconButton } from '@fluentui/react';
+import { getPropertyListItemIconWrapMoreStyles } from '../OATPropertyEditor.styles';
+import { useTranslation } from 'react-i18next';
+import { TemplateListItemListProps } from './TemplateListItem.types';
+
+export const TemplateListItem: React.FC<TemplateListItemListProps> = (
+    props
+) => {
+    const {
+        draggingTemplate,
+        item,
+        index,
+        deleteItem,
+        getDragItemClassName,
+        onDragEnter,
+        onDragEnterExternalItem,
+        onDragStart,
+        onMove,
+        onPropertyListAddition,
+        getSchemaText,
+        templatesLength
+    } = props;
+
+    // hooks
+    const { t } = useTranslation();
+
+    // state
+    const [subMenuActive, setSubMenuActive] = useState(false);
+
+    // callbacks
+    const addTopPropertyList = () => {
+        onPropertyListAddition(item);
+        setSubMenuActive(false);
+    };
+
+    // styles
+    const iconWrapMoreStyles = getPropertyListItemIconWrapMoreStyles();
+
+    return (
+        <div
+            id={`${item.name}_template_item`}
+            className={getDragItemClassName(index)}
+            key={index}
+            draggable
+            onDragStart={(e) => {
+                onDragStart(e, index);
+            }}
+            onDragEnter={
+                draggingTemplate
+                    ? (e) => onDragEnter(e, index)
+                    : () => onDragEnterExternalItem(index)
+            }
+        >
+            <Text>{item.displayName ? item.displayName : item.name}</Text>
+            <Text>{getSchemaText(item.schema)}</Text>
+
+            <IconButton
+                iconProps={{
+                    iconName: 'more'
+                }}
+                styles={iconWrapMoreStyles}
+                title={t('OATPropertyEditor.more')}
+                onClick={() => setSubMenuActive(!subMenuActive)}
+            >
+                {subMenuActive && (
```

Comment:
Is this WIP or should it be removed?

Reply:
Everything template related is currently deprecated. I think we are looking to bring htem back after private preview so I left that code hanging around for now.